lua/packer/neorocks.lua

```diff
@@ -0,0 +1,348 @@
+end
+
+neorocks.setup_paths = function(force)
+  local lua_version = neorocks._lua_version
+  local install_location = neorocks._hererocks_install_location
+
+  local match_install_path = neorocks._get_package_paths()
+
+  if force or not string.find(package.path, match_install_path, 1, true) then
+    package.path = match_install_path .. ';' .. package.path
+  end
+
+  local install_cpath = util.join_paths(install_location, "lib", "lua", lua_version.lua)
+  local match_install_cpath = string.format("%s/?.so", install_cpath)
+
+  if force or not string.find(package.cpath, match_install_cpath, 1, true) then
+    package.cpath = match_install_cpath .. ';' .. package.cpath
+  end
+end
+
+-- activate hererocks based on current $SHELL
+local function source_activate(install_location, activate_file)
+  return string.format('source %s', util.join_paths(install_location, 'bin', activate_file))
+end
+
+neorocks._source_string = function(install_location)
+  local user_shell = os.getenv("SHELL")
+  local shell = user_shell:gmatch("([^/]*)$")()
+  if shell == "fish" then
+    return source_activate(install_location, 'activate.fish')
+  elseif shell == "csh" then
+    return source_activate(install_location, 'activate.csh')
+  end
+  return source_activate(install_location, 'activate')
+end
+
+-- neorocks.get_async_installer = function(rocks, plugin_name, display)
+--   return async(function(self)
+    -- local new_env = setmetatable({}, {
+    --   __index = function(t, k)
+    --     local val
+    --     if k == 'io' then
+    --       val = setmetatable({
+    --         stdout = { write = function(_, s) display:task_update(plugin_name, string.format('stdout: %s', s)) end, },
+    --         stderr = { write = function(_, s) display:task_update(plugin_name, string.format('stderr: %s', s)) end, },
+    --       }, { __index = function() error('ACCESSING IO') end })
+    --     elseif k == 'os' then
+    --       val = setmetatable({exit = function() error('OS.EXIT') end}, {__index = os})
+    --     elseif k == 'print' then
+    --       val = function(...)
+    --         display:task_update(plugin_name, vim.inspect({...}))
+    --       end
+    --     else
+    --       val = _G[k]
+    --     end
+
+    --     t[k] = val
+    --     return t[k]
+    --   end
+    -- })
+
+    -- -- Install the package
+    -- local exec_luarocks = loadfile(util.join_paths(neorocks._hererocks_install_location, "bin", "luarocks"))
+    -- -- Redirect io.stdout & io.stderr
+    -- setfenv(exec_luarocks, new_env)('install', rocks[1])
+
+    -- TODO: I would really like to go back to the loadfile method and not shoot this out to a shell...
+    --          I'm worried about trying to compile multiple of these at the same time.
+    -- TODO: move to io.popen, so we can pipe the output to the display
+    -- vim.schedule_wrap(function()
+      -- TODO: Figure out how we could queue up these lua rocks commands to do in job start
```

Comment:
This is doable using https://github.com/wbthomason/packer.nvim/blob/a96f5f1ec20b758b27defdec433c664d57d48abe/lua/packer/async.lua#L60 with an `n` of 1 (for sequential execution).